homeassistant/components/fan/comfoconnect.py

```diff
@@ -0,0 +1,278 @@
+        return [SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
+
+    def turn_on(self, speed: str=None, **kwargs) -> None:
+        """Turn on the entity."""
+        if speed is None:
+            speed = SPEED_LOW
+        self.set_speed(speed)
+
+    def turn_off(self) -> None:
+        """Turn off the entity."""
+        self.set_speed(SPEED_OFF)
+
+    def set_speed(self, mode):
+        """Set fan speed."""
+        from pycomfoconnect.error import (
+            PyComfoConnectOtherSession, PyComfoConnectNotAllowed)
+        from pycomfoconnect.const import (
+            FAN_MODE_AWAY, FAN_MODE_LOW, FAN_MODE_MEDIUM,
+            FAN_MODE_HIGH
+        )
+
+        _LOGGER.debug('Changing fan mode to %s...' % mode)
+        try:
+            self._comfoconnect.connect(
+                self._token, self._friendly_name, self._pin)
+            _LOGGER.debug('Connected to bridge.')
+
+            if mode == SPEED_OFF:
+                self._comfoconnect.set_fan_mode(FAN_MODE_AWAY)
+            elif mode == SPEED_LOW:
+                self._comfoconnect.set_fan_mode(FAN_MODE_LOW)
+            elif mode == SPEED_MEDIUM:
+                self._comfoconnect.set_fan_mode(FAN_MODE_MEDIUM)
+            elif mode == SPEED_HIGH:
+                self._comfoconnect.set_fan_mode(FAN_MODE_HIGH)
+
+            # Update current mode
+            self._data[ATTR_SPEED] = mode
+            self.schedule_update_ha_state()
+
+        except PyComfoConnectOtherSession as ex:
+            _LOGGER.error('Another session with "%s" is active.',
+                          ex.devicename)
+
+        except PyComfoConnectNotAllowed:
+            _LOGGER.error('Could not register. Invalid PIN!')
+
+        finally:
+            self._comfoconnect.disconnect()
+            _LOGGER.debug('Disconnected from bridge.')
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        from pycomfoconnect import (
+            SENSOR_FAN_SPEED_MODE, SENSOR_FAN_SUPPLY_FLOW,
+            SENSOR_FAN_EXHAUST_FLOW, SENSOR_TEMPERATURE_EXTRACT,
+            SENSOR_TEMPERATURE_OUTDOOR, SENSOR_HUMIDITY_EXTRACT,
+            SENSOR_HUMIDITY_OUTDOOR
+        )
+
+        data = {}
+
+        for key, value in self._data.items():
+            if key == SENSOR_FAN_SPEED_MODE:
+                if value == 0:
+                    data[ATTR_SPEED] = SPEED_OFF
+                elif value == 1:
+                    data[ATTR_SPEED] = SPEED_LOW
+                elif value == 2:
+                    data[ATTR_SPEED] = SPEED_MEDIUM
+                elif value == 3:
+                    data[ATTR_SPEED] = SPEED_HIGH
+
+            elif key == SENSOR_HUMIDITY_EXTRACT:
+                data[ATTR_CURRENT_HUMIDITY] = value
+
+            elif key == SENSOR_HUMIDITY_OUTDOOR:
+                data[ATTR_OUTSIDE_HUMIDITY] = value
+
+            elif key == SENSOR_TEMPERATURE_EXTRACT:
+                data[ATTR_CURRENT_TEMPERATURE] = value / 10.0
```

Comment:
Wouldn't it make more sense to have these be added as a sensor platform? That way we get graphs and easy to automate against it.

- Move auth to `components/comfoconnect.py` and have it store the ComfoConnect object in `hass.data`
- add `sensor/comfoconnect.py` to expose sensors
- update the fan platform to use the ComfoConnect object in `hass.data`